### PR TITLE
Lesson 07 Character Device

### PR DESCRIPTION
**Pull request by danismi715**
- Create Makefile and initial driver code
- Add sysfs interface
- Add dev interface
- Add procfs interface
- Add entry point for module

Please, check my driver code. 
Thank you!

### CODE REVIEW

File: 02-KernelModules_CharacterDevice/my_chardev.c [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+static struct data_info{
+	int   crnt_size_buff;
+	int   size_buff;
+	char* data_buff;
+} dev_data_file;
+
+static struct proc_info{
+	int  data_length;
+	char data_buff[10];
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/my_chardev.c]: 10 bytes is not enough here: int value may require up to 9 decimal digits, but you've added in the proc_read() 2 more characters (a whitespace and \n).

File: 02-KernelModules_CharacterDevice/my_chardev.c [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+/**************************************************************************************************/
+
+
+/************************************** Interface for /proc ***************************************/
+
+static int proc_read(struct file *file_p, char __user *buffer, size_t length, loff_t *offset)
+{ 
+	int len = sprintf(proc_data_file.data_buff,"%d \n", dev_data_file.crnt_size_buff);
+    if( length < len ) return -EINVAL;
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/my_chardev.c]: Code indentation is broken here.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Plz use checkpatch (or indent) to fix indentation.

**Comment by danismi715** [comment]
- Expanded output buffer for procfs and fixed indents

**Comment by galkinletter** [comment]
fixed is correct

**Pull request merged**